src/lib/models/note.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const Model = require('./model');
+const validators = require('./validators');
+const TYPES = {
+  chargeVersion: 'charge_version'
+};
+
+class Note extends Model {
+  constructor (...args) {
+    super(...args);
+    this._type = TYPES.chargeVersion;
+  }
+
+  get noteId () { return this._noteId; }
+
+  set noteId (noteId) {
+    validators.assertId(noteId);
+    this._noteId = noteId;
+  }
+
+  get text () { return this._text; }
+
+  set text (text) {
+    validators.assertString(text);
+    this._text = text;
+  }
+
+  get typeId () { return this._typeId; }
+
+  set typeId (typeId) {
+    validators.assertId(typeId);
+    this._typeId = typeId;
+  }
+
+  get type () { return this._type; }
+
+  set type (type) {
+    validators.assertString(type);
+    this._type = type;
+  }
+
+  get licenceId () { return this._licenceId; }
+
+  set licenceId (licenceId) {
+    validators.assertNullableId(licenceId);
+    this._licenceId = licenceId;
+  }
+
+  get userId () { return this._userId; }
+
+  set userId (userId) {
```

Comment:
we could change this to be a user model instead of just the userId - but feasible to leave as is and extend later

Reply:
I did think about this and thought I should just go with id for now.  Maybe I'll leave so we can get this change in.